OpenTrons/opentrons-api
# feat(protocol-designer): add waste chute as option in move labware form

closes RAUT-746 and RAUT-717

# Overview

This PR adds waste chute as an option in the move labware drop down menu and adds a dismissible hint to let users know that once they move their labware to the waste chute, they can't get it back.

# Test Plan

NOTE: we do not have the asset to display the waste chute in the deck map yet!! So it will appear empty even when you have the waste chute in your protocol!

- go to settings and make sure your FF is turned on (the deck modification ff). If you don't see it, you'll have to enable the dev ffs (type `enablePrereleaseMode()` in console)
- create a flex protocol and add a waste chute and a gripper (note: the waste chute asset is a gripper for now)
- go to deck setup and 1 labware on the deck (make sure it is a reservoir or well plate or tiprack)
- create a move labware step and select the labware without the gripper. see that the new location drop down DOES NOT include the waste chute. Select the gripper and see that the dropdown DOES include the waste chute. Click to move it to the waste chute
- save the form and observe the modal hint saying they are disgarding their labware. click on the checkbox to permanently dismiss it
- repeat the move labware step with the tiprack, moving it to the waste chute, and see that when you add it, the hint modal is not there.
- mess around with move labware and the new location dropdown, the waste chute should only be available if you are moving with the gripper, the labware is on deck, and the waste chute is present.

The hint modal:
<img width="652" alt="Screen Shot 2023-09-20 at 3 26 16 PM" src="https://github.com/Opentrons/opentrons/assets/66035149/06c9b147-4efa-4dae-bfc2-6943a192a900">

# Changelog

- add copy to the move labware header for moving to a waste chute
- add logic to the `getUnocuppiedLabwareLocationOptions` util for waste chute option
- add a hint for discarding of the waste chute

# Review requests

see test plan

# Risk assessment

low - shouldn't be able to access anything without the feature flag turned on.
